### PR TITLE
Testing branch

### PR DESCRIPTION
**Pull request by blazicmarko**
Lots of renaming. Filter is request model, but first I putted in DB models, so I renamed that.
IdExistsException was no longer needed, so it's deleted.
All of test classes got DB in name (my mistake), so it needed to be renamed
I commit all test for Movies and Projections (service, resource, etc.).

### CODE REVIEW

File: Cinema/src/main/java/com/example/cinema/service/MoviesService.java [hunk 1]
```diff
@@ -32,10 +32,6 @@ public LocalTime findTime(ProjectionDB projection) {
         return moviesMapper.findTime(projection);
     }
 
-    public Integer getLastId() {
-        return InitService.getMovieLastId();
-    }
-
```
Comment (blazicmarko) [inline Cinema/src/main/java/com/example/cinema/service/MoviesService.java:38]: Unused method. Deleted

File: Cinema/src/main/java/com/example/cinema/model/requestModel/MovieUpdateReq.java [hunk 2]
```diff
@@ -26,6 +26,15 @@
...
 
+    public MovieUpdateReq(Integer id, String name, @Min(value = 1, groups = SecondPriorGroup.class) @Max(value = 10, groups = SecondPriorGroup.class) Double grade, @Min(value = 1900, groups = SecondPriorGroup.class) @Max(value = 2021, groups = SecondPriorGroup.class) Integer year, Integer idGenre, LocalTime time) {
+        this.id = id;
+        this.name = name;
+        this.grade = grade;
+        this.year = year;
+        this.idGenre = idGenre;
+        this.time = time;
+    }
+
```
Comment (blazicmarko) [inline Cinema/src/main/java/com/example/cinema/model/requestModel/MovieUpdateReq.java]: Constructor for purpose of testing

File: Cinema/src/test/java/com/example/cinema/service/ProjectionDBServiceTest.java [hunk 3]
```diff
@@ -142,11 +142,17 @@ public void UpdateReturningRightData() {
     }
 
     @Test
-    public void UpdateReturnNoIdException(){
```
Comment (blazicmarko) [inline Cinema/src/test/java/com/example/cinema/service/ProjectionDBServiceTest.java:145]: To make difference between Null and Zero. This test is for NoIdException and it throws in two ifs

File: Cinema/src/main/java/com/example/cinema/exception/ApiExceptionHandler.java [hunk 4]
```diff
@@ -30,17 +30,6 @@
         return new ResponseEntity<>(basicResponse, badRequest);
     }
 
-    @ExceptionHandler(value = {IdExistsException.class})
```
Comment (blazicmarko) [inline Cinema/src/main/java/com/example/cinema/exception/ApiExceptionHandler.java:33]: Deleted handler for IdExsitsException

File: Cinema/src/main/java/com/example/cinema/model/requestModel/ProjectionUpdateReq.java [hunk 5]
```diff
@@ -23,6 +23,8 @@
     private LocalTime startTime;
     private LocalTime endTime;
 
+    public ProjectionUpdateReq() {
+    }
```
Comment (blazicmarko) [inline Cinema/src/main/java/com/example/cinema/model/requestModel/ProjectionUpdateReq.java:27]: Constructor for purpose of testing

File: Cinema/src/main/java/com/example/cinema/model/requestModel/MovieUpdateReq.java (hunk 2 above)
Comment (igorcirovic) [reply Cinema/src/main/java/com/example/cinema/model/requestModel/MovieUpdateReq.java]: Try to avoid adding constructors and methods, if they are only used to enable unit tests. Same thing applies to access modifiers (public, private, protected), try to avoid changing them just because of unit tests.

File: Cinema/src/main/java/com/example/cinema/model/requestModel/ProjectionUpdateReq.java (hunk 5 above)
Comment (igorcirovic) [reply Cinema/src/main/java/com/example/cinema/model/requestModel/ProjectionUpdateReq.java:27]: Same as above.

File: Cinema/src/test/java/com/example/cinema/resource/ProjectionResourceTest.java [hunk 6]
```diff
@@ -13,13 +18,18 @@
 import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
 
+import java.time.LocalDate;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
```
Comment (igorcirovic) [inline Cinema/src/test/java/com/example/cinema/resource/ProjectionResourceTest.java:25]: Avoid these. Always replace these with individual imports, to make sure you are only importing resources you need.

File: Cinema/src/test/java/com/example/cinema/service/MovieServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    MoviesService moviesService;
+
+    @MockBean
+    MoviesMapper moviesMapper;
+
+    @MockBean
+    GenreService genreService;
+
+    @Test
+    void findTime() {
```
Comment (igorcirovic) [inline Cinema/src/test/java/com/example/cinema/service/MovieServiceTest.java:40]: Add edge case coverage for findTime, findAll, insert, update, getByName.

File: Cinema/src/main/java/com/example/cinema/model/requestModel/MovieUpdateReq.java (hunk 2 above)
Comment (blazicmarko) [reply Cinema/src/main/java/com/example/cinema/model/requestModel/MovieUpdateReq.java]: All constructors are now zero argument constructor. For purpose of program I didn't need any other constructor. Fixed

File: Cinema/src/main/java/com/example/cinema/model/requestModel/ProjectionUpdateReq.java (hunk 5 above)
Comment (blazicmarko) [reply Cinema/src/main/java/com/example/cinema/model/requestModel/ProjectionUpdateReq.java:27]: Fixed

File: Cinema/src/test/java/com/example/cinema/resource/ProjectionResourceTest.java (hunk 6 above)
Comment (blazicmarko) [reply Cinema/src/test/java/com/example/cinema/resource/ProjectionResourceTest.java:25]: Fixed.

File: Cinema/src/test/java/com/example/cinema/service/MovieServiceTest.java (hunk 7 above)
Comment (blazicmarko) [reply Cinema/src/test/java/com/example/cinema/service/MovieServiceTest.java:40]: Created
